patroni/__main__.py

```diff
@@ -44,9 +45,16 @@ def __init__(self, config: 'Config') -> None:
 
     def load_dynamic_configuration(self) -> None:
         from patroni.exceptions import DCSError
+
         while True:
             try:
                 cluster = self.dcs.get_cluster()
+                if cluster:
+                    same_name = [member for member in cluster.members if member.name == self.config['name']]
+                    if len(same_name) > 0 and any(member.state == 'running' for member in same_name):
```

Comment:
Actually, the check should be more sophisticated than just `state == 'running'`.

Let's imagine that Patroni was killed with "-9" signal and init system immediately restarts it.
With this change Patroni will refuse to start until the member key expires.

We should at least also check that start for a found member `api_url` and `conn_url` are not the same.
Or, in case if they are the same try to call `api_url` and don't error out if it can't be accessed.

Reply:
This makes sense, I missed the `get_member` function, thanks for pointing out. Implemented this and moved the logic to a separate function that would be called after `requests` is initialized but before the first run